system-contract-dapp-playground/src/components/navbar/index.tsx

```diff
@@ -23,29 +23,146 @@
 import Link from 'next/link';
 import Image from 'next/image';
 import { motion } from 'framer-motion';
+import WalletPopup from '../wallet-popup';
+import { useToast } from '@chakra-ui/react';
+import { useEffect, useState } from 'react';
+import { usePathname } from 'next/navigation';
+import { BsChevronDown } from 'react-icons/bs';
+import { NetworkName } from '@/types/interfaces';
+import { isProtectedRoute } from '@/utils/helpers';
+import { CommonErrorToast } from '../toast/CommonToast';
+import { loadAccountInfoFromCookies } from '@/api/cookies';
 import { navVariants } from '@/libs/framer-motion/variants';
 
 const Navbar = () => {
+  // local states
+  const toaster = useToast();
+  const pathname = usePathname();
+  const [accounts, setAccounts] = useState<string[]>([]);
+  const [isConnected, setIsConnected] = useState(false);
+  const [didWalletPop, setDidWalletPop] = useState(false);
+  const [network, setNetwork] = useState<NetworkName>('testnet');
+
+  // listen to pathname change event to retrieve account information cookies
+  useEffect(() => {
+    if (isProtectedRoute(pathname)) {
+      // retrieve account info from cookies
+      const accountsInfo = loadAccountInfoFromCookies();
+
+      // just best pratice to handle error as this would rarely happen as we have already gated all the checks in middleware.js
+      if (
+        !accountsInfo.isConnected ||
+        !accountsInfo.accounts ||
+        !accountsInfo.network ||
+        accountsInfo.error
+      ) {
+        CommonErrorToast({
```

Comment:
Q: does this log anything?
IF not should be be logging `accountInfo` to help troubleshoot

Reply:
yes it does log the error in the client-side console. I should change the message from "Unknown error appeared..." to "'See client console for more information'" though.